src/VestRewardsDistributionJob.sol

```diff
@@ -0,0 +1,178 @@
+        if (!distributions.add(rewDist)) revert RewardDistributionExists(rewDist);
+        if (delay < minimumDelay) revert LessThanMinimumDelay(delay);
+        distributionDelays[rewDist] = delay;
+        emit AddRewardDistribution(rewDist, delay);
+    }
+
+    function removeRewardDistribution(address rewDist) external auth {
+        if (!distributions.remove(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        delete distributionDelays[rewDist];
+        emit RemoveRewardDistribution(rewDist);
+    }
+
+    function modifyDistributionDelay(address rewDist, uint256 delay) external auth {
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        if (delay < minimumDelay) revert LessThanMinimumDelay(delay);
+        distributionDelays[rewDist] = delay;
+        emit ModifiedDistributionDelay(rewDist, delay);
+    }
+
+    function work(bytes32 network, bytes calldata args) public {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+
+        (address rewDist) = abi.decode(args, (address));
+
+        // prevent keeper from calling random contracts having distribute()
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        // ensure that the right delay has elapsed
+        if (canDistributeAfter(rewDist) >= block.timestamp) revert CannotDistributeYet(rewDist);
+        // we omit checking the unpaid amount because if it is 0 it will revert during distribute()
+        uint256 distAmount = VestedRewardsDistributionLike(rewDist).distribute();
+        emit Work(network, rewDist, distAmount);
+    }
+
+    function workable(bytes32 network) external view override returns (bool, bytes memory) {
+        if (!sequencer.isMaster(network)) return (false, bytes("Network is not master"));
+
+        uint256 distributionsLen = distributions.length();
+        if (distributionsLen > 0) {
+            address distributable;
+            // this is used to find the distribute() than could have been called the earliest
+            // we use a hack that ensures the right functionality while avoiding an extra check for 0 value
+            uint256 earliestDistCall = block.timestamp + timeMagicNumber;
+            for (uint256 i = 0; i < distributionsLen; i++) {
+                address rewDist = distributions.at(i);
+                uint256 nextDistCall = canDistributeAfter(rewDist);
+                // timestamp should be strictly greater than nextDistCall
+                if (nextDistCall < block.timestamp) {
+                    uint256 vestId = VestedRewardsDistributionLike(rewDist).vestId();
+                    DssVestWithGemLike dssVest = VestedRewardsDistributionLike(rewDist).dssVest();
+                    uint256 amount = dssVest.unpaid(vestId);
+                    if (amount > 0){
+                        if (nextDistCall < earliestDistCall)
+                            distributable = rewDist;
```

Comment:
Shouldn't you just return here?

Otherwise, you will keep iterating over the list and if any job matches the criteria, it will be returned instead, so it would behave in a last-in/first-out fashion. It's not a problem on itself, but it can be confusing.

Reply:
the above algo is looking for the minimum so we have to go through the whole list